src/server/api/boards/boards.js

```diff
@@ -19,10 +19,19 @@ function createFindQuery(username) {
 }
 
 function idQuery(username, id) {
-  return {
-    'username': username,
-    '_id': id,
-  };
+  try {
+    let mongoId = new mongodb.ObjectId(id);
+
+    return {
+      'username': username,
+      '_id': mongoId,
+    };
+  } catch (error) {
+    return {
```

Comment:
i might say that the it should be `return null;` instead of `return { username: username, _id: null };` but discuss it with the team.